design_pim/overview.rst

```diff
@@ -0,0 +1,327 @@
+
+The real thing
+++++++++++++++
+
+Here is how to do it using RequireJS:
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/locale_logger.js
+    define(
+        ['pim/user-context'],
+        function (UserContext) {
+            return {
+                logLocale: function () {
+                    console.log(UserContext.get('uiLocale'));
+                }
+            }
+        }
+    )
+
+
+The ``define`` method takes two arguments: the first one is an array of the dependencies we want to use (``pim/user-context`` in this case) and a callback that will be called every time our module will be requested. This callback will receive as argument the list dependencies we asked for. Note that you can name the arguments of the callback as you want.
+
+Register it
++++++++++++
+
+Now that we have this small module, we would like to use it somewhere else. To do so, we need to register it in the requirejs configuration file.
+
+.. code-block:: yaml
+
+    # src/Acme/Bundle/CustomBundle/Resources/config/requirejs.yml
+    config:
+        paths:
+            my_locale_logger: acmecustom/js/locale_logger
+
+
+We just declared a new module named ``my_locale_logger`` located in ``web/bundles/acmecustom/js/locale_logger.js``.
+Note that we don't need to specify the file extension and it is located in the public web folder (moved during the ``bin/console assets:install`` command execution). Also, ``my_locale_logger`` name is a free name and you can replace it with whatever you want.
+
+Make it available
++++++++++++++++++
+
+To make your module available you need to clear your cache:
+
+    rm -rf ./var/cache/*
+
+dump the assets in the web folder
+
+    bin/console assets:install web
+
+You can also decide to symlink those assets instead of copying them:
+
+    bin/console assets:install --symlink web
+
+After that the last step is to build the bundle.js file
+
+    npm run webpack
+
+This command will compile and minify all the pim files into web files and dump them in the public folder.
+
+You can also use the watch command which will recompile this file each time you modify a registered module.
+
+    npm run webpack-watch
+
+
+Summary
++++++++
+
+We just created our first module and made it available for other part of the application. You can now use it by requirering ``my_locale_logger``.
+
+
+Create our first backbone view
+==============================
+
+Know that we know how to create a requirejs module, let's create our first Backbone view.
+
+.. code-block:: javascript
+    :linenos:
+
+    // src/Acme/Bundle/CustomBundle/Resources/public/js/new_view.js
+    define(
+        ['backbone'],
+        function (Backbone) {
+            return Backbone.View.extend({
+
+            });
+        }
+    );
+
+
+This is the simplest view we can create using backbone. It will result into an empty div if we add it to the DOM.
```

Comment:
The first sentence is weird, what do you mean ?

Reply:
I don't know: this is the simplest view we can create using backbone framework. I don't understand what you don't understand :/